payment/datastore.go

```diff
@@ -300,19 +302,17 @@ func (pg *Postgres) GetOrderCreds(orderID uuid.UUID) (*[]OrderCreds, error) {
 	return nil, nil
 }
 
-// GetOrderCredsByItemID returns the order credentials for a OrderID
-func (pg *Postgres) GetOrderCredsByItemID(orderID uuid.UUID) (*[]OrderCreds, error) {
-	orderCreds := []OrderCreds{}
-	err := pg.DB.Select(&orderCreds, "select * from order_creds where order_id = $1 and signed_creds is not null", orderID)
-	if err != nil {
+// GetOrderCredsByItemID returns the order credentials for a OrderID by the itemID
+func (pg *Postgres) GetOrderCredsByItemID(orderID uuid.UUID, itemID uuid.UUID) (*OrderCreds, error) {
+	orderCreds := OrderCreds{}
+	err := pg.DB.Get(&orderCreds, "select * from order_creds where order_id = $1 and id = $2 and signed_creds is not null", orderID, itemID)
+	if err == sql.ErrNoRows {
+		return nil, nil
```

Comment:
might be good to propagate this as a wrapped "error" so we can 404 based on the error instead of the lack of orderCreds?

Reply:
Since this returning nil, it will give a 404, but updated the message - https://github.com/brave-intl/bat-go/pull/302/commits/0534a458ca2e6a994374f5a69d490f0f43b35534

https://github.com/brave-intl/bat-go/blob/bad4ef2ddbb2da4171cebcfe9c8ac146e4742e0b/payment/controllers.go#L369-L375